src/Platform/Microsoft.Testing.Platform/OutputDevice/Terminal/TerminalTestReporter.cs

```diff
@@ -704,6 +704,61 @@ internal void WriteWarningMessage(string assembly, string? targetFramework, stri
     internal void WriteErrorMessage(string assembly, string? targetFramework, string? architecture, Exception exception)
         => WriteErrorMessage(assembly, targetFramework, architecture, exception.ToString());
 
-    internal void WriteMessage(string text)
-        => _terminalWithProgress.WriteToTerminal(terminal => terminal.AppendLine(text));
+    internal void WriteMessage(string text, SystemConsoleColor? color = null)
+    {
+        if (color != null)
+        {
+            _terminalWithProgress.WriteToTerminal(terminal =>
+            {
+                terminal.SetColor(ToTerminalColor(color.ConsoleColor));
+                terminal.AppendLine(text);
+                terminal.ResetColor();
```

Comment:
Shall we be using some kind of try/finally pattern here?

Reply:
I don't think there is much opportunity for error. It is probably much easier to just reset the color at the end of the run, if you are afraid of keeping the color set to non-default. But so far I haven't seen it happen.

Reply:
Ok trusting you here.